dali/c_api/c_api.cc

```diff
@@ -283,26 +284,36 @@ int daliGetMaxBatchSize(daliPipelineHandle_t pipe_handle) {
   return (*pipe_handle)->pipeline->max_batch_size();
 }
 
+int daliInputFeedCount(daliPipelineHandle_t pipe_handle, const char *input_name) {
+  auto &pipeline = (*pipe_handle)->pipeline;
+  return pipeline->InputFeedCount(input_name);
+}
+
+void daliPrefetch(daliPipelineHandle_t pipe_handle) {
+  auto &pipeline = (*pipe_handle)->pipeline;
+  pipeline->Prefetch();
+}
 
 void daliPrefetchUniform(daliPipelineHandle_t pipe_handle, int queue_depth) {
   auto &pipeline = (*pipe_handle)->pipeline;
-  for (int i = 0; i < queue_depth; ++i) {
-    pipeline->RunCPU();
-    pipeline->RunGPU();
+  auto sz = pipeline->GetQueueSizes();
+  if (queue_depth != sz.cpu_size || queue_depth != sz.gpu_size) {
+    DALI_WARN("daliPrefetchUniform is deprecated and setting queue_length different than"
+    " the one set in the pipeline has no effect. Use daliPrefetch instead.");
   }
+  pipeline->Prefetch();
 }
 
 
 void daliPrefetchSeparate(daliPipelineHandle_t pipe_handle,
                           int cpu_queue_depth, int gpu_queue_depth) {
   auto &pipeline = (*pipe_handle)->pipeline;
-  for (int i = 0; i < gpu_queue_depth; ++i) {
-    pipeline->RunCPU();
-    pipeline->RunGPU();
-  }
-  for (int i = 0; i < cpu_queue_depth; ++i) {
-    pipeline->RunCPU();
+  auto sz = pipeline->GetQueueSizes();
+  if (cpu_queue_depth != sz.cpu_size || gpu_queue_depth != sz.gpu_size) {
+    DALI_WARN("daliPrefetchUniform is deprecated and setting queue_length different than"
```

Comment:
Yes, it's normally only called once per the lifetime of the pipeline. WARN_ONCE has a downside that it would be only once per the lifetime of the process.